app/src/main/kotlin/com/wire/android/feature/conversation/Conversation.kt

```diff
@@ -1,3 +1,19 @@
 package com.wire.android.feature.conversation
 
-data class Conversation(val id: String, val name: String? = null, val type: ConversationType)
+import android.os.Parcelable
+import com.wire.android.core.extension.EMPTY
+import kotlinx.android.parcel.Parcelize
+
+data class Conversation(
+    val id: ConversationID,
+    val name: String? = null,
+    val type: ConversationType
+)
+
+@Parcelize
+data class ConversationID(val value: String, val domain: String) : Parcelable {
```

Comment:
Core models should not depend on `android.os` things.

If this is sused for Bundles, perhaps we should deconstruct the ID and actually put `value` and `domain` separately in the bundle.